client_code/_Components/ButtonMenu/__init__.py

```diff
@@ -13,380 +13,275 @@
-    self.menu_button.background_color = value
-
-  @anvil_prop
-  @property
-  def button_text_color(self, value) -> str:
-    """The colour of the text on the Button."""
-    self.menu_button.text_color = value
-
-  @anvil_prop
-  @property
-  def button_font_size(self, value) -> int:
-    """The font size of the text displayed on the Button."""
-    self.menu_button.font_size = value
-
-  @anvil_prop
-  @property
-  def icon(self, value) -> str:
-    """The icon to display on the Button."""
-    self.menu_button.icon = value
-
-  @anvil_prop
-  @property
-  def icon_color(self, value) -> str:
-    """The colour of the icon displayed on the Button."""
-    self.menu_button.icon_color = value
-
-  @anvil_prop
-  @property
-  def icon_size(self, value) -> int:
-    """The size (pixels) of the icon displayed on this component."""
-    self.menu_button.icon_size = value
-
-  @anvil_prop
-  @property
-  def icon_position(self, value) -> str:
-    """The alignment of the icon on this component."""
-    self.menu_button.icon_position = value
-
-  @anvil_prop
-  @property
-  def spacing(self, value) -> list:
-    """The margin and padding (pixels) of the component."""
-    self.menu_button.spacing = value
-
-  @anvil_prop
-  @property
-  def align(self, value) -> str:
-    """The position of this component in the available space."""
-    self.menu_button.dom_nodes['anvil-m3-button'].classList.toggle(
-      'anvil-m3-full-width', False
-    )
-    self.menu_button.dom_nodes['anvil-m3-button-component'].style.removeProperty(
-      'justify-content'
+class ButtonMenu(ButtonMenuTemplate, MenuMixin):
+    def __init__(self, **properties):
+        self.tag = ComponentTag()
+        self._props = properties
+        self._design_name = ""
+        self._cleanup = noop
+        self._open = False
+        self._hoverIndex = None
+        self._children = None
+        self._shown = False
+        self._itemIndices = set()
+        self._menuNode = self.dom_nodes['anvil-m3-buttonMenu-items-container']
+        self._btnNode = get_dom_node(self.menu_button).querySelector("button")
+
+        self.init_components(**properties)
+    
+        self.add_event_handler("x-anvil-page-added", self._on_mount)
+        self.add_event_handler("x-anvil-page-removed", self._on_cleanup)
+
+    def _on_mount(self, **event_args):
+        self._shown = True
+        document.addEventListener('keydown', self._call_handle_keyboard_events)
+        self._btnNode.addEventListener('click', self._handle_click)
+        self._menuNode.addEventListener('click', self._handle_child_clicked)
+        document.addEventListener('click', self._handle_body_click)
+        
+        # We still have a reference to the dom node but we've moved it to the body
+        # This gets around the fact that Anvil containers set their overflow to hidden
+        document.body.append(self._menuNode)
+        super()._setup_fui(self._btnNode, self._menuNode)
```

Comment:
```suggestion
        self._setup_fui(self._btnNode, self._menuNode)
```

this doesn't need to be a super